src/App.tsx

```diff
@@ -1,61 +1,134 @@
 import './App.scss';
+import { useState } from 'react';
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import { Todo } from './services/types';
+
+function getUserById(id: number) {
+  return usersFromServer.find(user => user.id === id);
+}
+
+const todosWithUser = (): Todo[] => {
+  return todosFromServer.map(todo => {
+    return {
+      ...todo,
+      user: getUserById(todo.id),
+    };
+  });
+};
+
+const titlePatter = /[^a-zA-ZА-ЩЬЮЯҐЄІЇа-щьюяґєії\d\s]/gm;
 
 export const App = () => {
+  const [todos, setTodos] = useState<Todo[]>(todosWithUser);
+
+  const [title, setTitle] = useState('');
+  const [isTitleValid, setIsTitleValid] = useState(true);
+
+  const [user, setUser] = useState(0);
+  const [isUserValid, setIsUserValid] = useState(true);
+
+  const [completed] = useState(false);
+
+  const reset = () => {
+    setTitle('');
+    setUser(0);
+  };
+
+  const handleSubmit = (event: React.FormEvent) => {
+    event.preventDefault();
+    let abortSubmit = false;
+
+    if (!title) {
+      setIsTitleValid(false);
+      abortSubmit = true;
```

Comment:
just use `return;` instead of creating an additional variable
```suggestion
      return;
```

Reply:
I used a variable to return only after checking both fields